project_classification/__init__.py

```diff
@@ -0,0 +1,82 @@
+# -*- coding: utf-8 -*-
+##############################################################################
+#
+#    Author: Joël Grand-guillaume (Camptocamp)
+#    Copyright 2011 Camptocamp SA
+#
+#    This program is free software: you can redistribute it and/or modify
+#    it under the terms of the GNU Affero General Public License as
+#    published by the Free Software Foundation, either version 3 of the
+#    License, or (at your option) any later version.
+#
+#    This program is distributed in the hope that it will be useful,
+#    but WITHOUT ANY WARRANTY; without even the implied warranty of
+#    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#    GNU Affero General Public License for more details.
+#
+#    You should have received a copy of the GNU Affero General Public License
+#    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+##############################################################################
+from . import project_classification
+import psycopg2
+
+
+def fill_classification_id(cr):
+        query_nb_records = 'SELECT count(id) from project_project;'
+        cr.execute(query_nb_records)
+        res = cr.fetchall()
+        if res[0][0] != 0:
+            try:
+                query_classification_id = (
+                    'SELECT count(id) from project_project '
+                    'where classification_id is null;'
+                    )
+                cr.execute(query_classification_id)
+                res2 = cr.fetchall()
+                if res[0][0] == res2[0][0]:
+                    # all project have a classification_id set
+                    return
+
+            except psycopg2.ProgrammingError:
+                cr.rollback()
```

Comment:
What do you expect to rollback? It's gonna blow your database if you rollback what has been done before the init_hook is executed (anyway the `except` could be removed as proposed in my other comment).